src/transformer/descriptor/identifier/identifier.ts

```diff
@@ -1,8 +1,10 @@
 import * as ts from 'typescript';
+import { IScope } from '../../scope/scope.interface';
 import { GetDescriptor } from '../descriptor';
 import { TypescriptHelper } from '../helper/helper';
 
-export function GetIdentifierDescriptor(node: ts.Identifier): ts.Expression {
+export function GetIdentifierDescriptor(node: ts.Identifier, scope: IScope): ts.Expression {
     const declaration: ts.Declaration = TypescriptHelper.GetDeclarationFromNode(node);
-    return GetDescriptor(declaration);
+    return GetDescriptor(declaration, scope);
+    return GetDescriptor(declaration, scope);
```

Comment:
It seems you have a repetition here

Reply:
Thanks, However for simplicity, I've included these changes in the generic refactoring PR because the cache will disappear completely.